frontend/src/services/search/SearchFilterManager.ts

```diff
@@ -0,0 +1,46 @@
+import { FilterOption } from "../../components/search/SearchFilterAccordion/SearchFilterAccordion";
+import { QueryParamKey } from "../../types/searchTypes";
+
+type UpdateQueryParamsFunction = (
+  checkedSet: Set<string>,
+  queryParamKey: QueryParamKey,
+) => void;
+
+export default class SearchFilterManager {
+  options: FilterOption[];
+  setOptions: React.Dispatch<React.SetStateAction<FilterOption[]>>;
+  updateQueryParams: UpdateQueryParamsFunction;
+  formRef: React.RefObject<HTMLFormElement>;
+
+  constructor(
+    options: FilterOption[],
+    setOptions: React.Dispatch<React.SetStateAction<FilterOption[]>>,
+    updateQueryParams: UpdateQueryParamsFunction,
+    formRef: React.RefObject<HTMLFormElement>,
+  ) {
+    this.options = options;
+    this.setOptions = setOptions;
+    this.updateQueryParams = updateQueryParams;
+    this.formRef = formRef;
+  }
+
+  static initializeOptions = (
+    initialFilterOptions: FilterOption[],
+    initialQueryParams: string | null,
+  ) => {
+    // convert the request URL query params to a set
+    const initialParamsSet = new Set(
+      initialQueryParams ? initialQueryParams.split(",") : [],
+    );
+    return initialFilterOptions.map((option) => ({
+      ...option,
+      isChecked: initialParamsSet.has(option.value),
+      children: option.children
+        ? option.children.map((child) => ({
+            ...child,
+            isChecked: initialParamsSet.has(child.value),
+          }))
+        : undefined,
+    }));
+  };
+}
```

Comment:
Start the creation of this class (currently not used but I think we should start to move some of the management of checked options here (how many total checked, how many in a section are checked, if all in a section are checked are not) - to make it easy to just grab these values instead of looping through all the options everytime. 

It probably won't lead to a ton of performance improvement though - just more for our own sanity :)

Reply:
Might try this in a part 2 PR on this #1491 ticket

Reply:
Cool. Should we put a TODO in this file noting as much (that it's not being used yet and a link to a ticket to implement/explore)?

Reply:
yea good catch - will add that!